### PR TITLE
gitoxide: 0.19.0 -> 0.20.0

### PR DESCRIPTION
**Pull request by yanganto**
###### Description of changes
Update gitoxide

###### Things done
- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by syberant — APPROVED** [review]

**Comment by syberant** [comment]
Thanks for updating!

**Comment by syberant** [comment]
Wait, I see now that it doesn't work on aarch64-darwin and x86_64-darwin anymore while [it did last time](https://github.com/NixOS/nixpkgs/pull/202988/checks?check_run_id=9718586056):
```
       >    Compiling gitoxide v0.20.0 (/private/tmp/nix-build-gitoxide-0.20.0.drv-0/source)
       > error: linking with `/nix/store/da41skz6xclm0v9gl88q34jkhfp0zx6b-clang-wrapper-11.1.0/bin/cc` failed: exit status: 1
       >   |
       >   = note: "/nix/store/da41skz6xclm0v9gl88q34jkhfp0zx6b-clang-wrapper-11.1.0/bin/cc" "-arch" "x86_64" "-m64" "/private/tmp/nix-build-gitoxide-0.20.0.drv-0/rustceQWAuX/symbols.o" "/private/tmp/nix-build-gitoxide-0.20.0.drv-0/source/target/x86_64-apple-darwin/release/deps/gix-d464f8444b3a7396.gix.2fc604d1-cgu.0.rcgu.o" "-L" "/private/tmp/nix-build-gitoxide-0.20.0.drv-0/source/target/x86_64-apple-darwin/release/deps" "-L" "/private/tmp/nix-build-gitoxide-0.20.0.drv-0/source/target/release/deps" "-L" "/private/tmp/nix-build-gitoxide-0.20.0.drv-0/source/target/x86_64-apple-darwin/release/build/libz-sys-3455178bb9658416/out/lib" "-L" "/private/tmp/nix-build-gitoxide-0.20.0.drv-0/source/target/x86_64-apple-darwin/release/build/sha1-asm-b9f0f1be1b46fc8a/out" "-L" "/nix/store/jm0hjqf69frydhbsicw2is6bkkqqvqqb-rustc-1.65.0/lib/rustlib/x86_64-apple-darwin/lib" "/private/tmp/nix-build-gitoxide-0.20.0.drv-0/rustceQWAuX/liblibz_sys-9b7ab3ba36b04ec4.rlib" "/private/tmp/nix-build-gitoxide-0.20.0.drv-0/rustceQWAuX/libsha1_asm-b9bbaa226c131451.rlib" "/nix/store/jm0hjqf69frydhbsicw2is6bkkqqvqqb-rustc-1.65.0/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-3097cb3d32beb6d3.rlib" "-lcurl" "-liconv" "-lSystem" "-lresolv" "-lc" "-lm" "-liconv" "-L" "/nix/store/jm0hjqf69frydhbsicw2is6bkkqqvqqb-rustc-1.65.0/lib/rustlib/x86_64-apple-darwin/lib" "-o" "/private/tmp/nix-build-gitoxide-0.20.0.drv-0/source/target/x86_64-apple-darwin/release/deps/gix-d464f8444b3a7396" "-Wl,-dead_strip" "-nodefaultlibs"
       >   = note: ld: library not found for -lcurl
       >           clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
       >           
       >
       > error: could not compile `gitoxide` due to previous error
       For full logs, run 'nix log /nix/store/5s11jnl88f15kmmw8xbby3fvkyblqydy-gitoxide-0.20.0.drv'.
```

I don't have a MacOS machine to debug this unfortunately and can only confirm that it does build and work correctly on my NixOS installation.
There is an [alternative `reqwest` backend](https://github.com/Byron/gitoxide/discussions/666) that doesn't use `curl` but upstream has not switched to it because it is feature incomplete.

*Maybe* relevant: https://github.com/Byron/gitoxide/issues/602

**Comment by syberant** [comment]
@stephank, I see you have fixed the build for darwin once before. Do you perhaps have any idea why this happens?

**Comment by stephank** [comment]
Tbh, I'm more surprised it works on Linux. The curl dependency is not Darwin-specific. Perhaps `curl.dev` is propagated from somewhere on Linux?

I added a commit here that builds & runs on aarch64-darwin: https://github.com/NixOS/nixpkgs/compare/master...stephank:nixpkgs:gitoxide-0.20

**Comment by syberant** [comment]
Hmm, thank you for pointing out the missing curl dependency. I suppose it is propagated from somewhere (though I haven't been able to pin down from where exactly) and it used to work on darwin too as 0.19.0 still passed on darwin while also needing curl. With this in mind it might make more sense to have it as an explicit dependency for both Linux and MacOS.

As an aside I'm curious why you deleted the `libiconv Security SystemConfiguration` dependencies, I believe `Security` in particular used to be needed to satisfy the openssl requirement.
As the openssl package is only added as a dependency for non-darwin systems I'm not sure why it works for you.

**Comment by stephank** [comment]
I only did basic testing (`--help`), so maybe those are still needed. I was just curious if they were still necessary to build. If you're concerned they may break some runtime functionality, maybe just leave them in until someone comes along to verify things. 🙂

**Review by syberant — CHANGES_REQUESTED** [review]
Alright, leaving them sounds good. Thanks for your help!

Then concretely I propose the following `buildInputs` which should work on darwin as well:
```nix
  buildInputs = [ curl ] ++ (if stdenv.isDarwin
    then [ libiconv Security SystemConfiguration ]
    else [ openssl ]);
```

**Comment by syberant** [comment]
@yanganto, do you have time to commit (and squash) these changes?

**Comment by yanganto** [comment]
Thank for mention me, if anything I forgot to do.  Please kindly let me know. :pray:

File: pkgs/applications/version-management/gitoxide/default.nix [hunk 1]
```diff
@@ -4,29 +4,27 @@
 , cmake
 , pkg-config
 , stdenv
-, libiconv
-, Security
-, SystemConfiguration
```
Comment (syberant) [inline pkgs/applications/version-management/gitoxide/default.nix:9]: We're still using these for darwin in the `buildInputs` (at least until someone can properly verify that `gitoxide` works without), so we shouldn't delete them. OfBorg is failing on this.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -1626,9 +1626,7 @@ with pkgs;
 
   gita = python3Packages.callPackage ../applications/version-management/gita { };
 
-  gitoxide = callPackage ../applications/version-management/gitoxide {
-    inherit (darwin.apple_sdk.frameworks) Security SystemConfiguration;
-  };
```
Comment (syberant) [inline pkgs/top-level/all-packages.nix:1631]: Same here.

**Review by syberant — CHANGES_REQUESTED** [review]

File: pkgs/applications/version-management/gitoxide/default.nix (hunk 1 above)
Comment (yanganto) [reply pkgs/applications/version-management/gitoxide/default.nix:9]: Oh.  My bad. Sorry about forgetting to check the CI again. Thank you for your kind notes here. :pray:

**Review by syberant — APPROVED** [review]
Looks good, it still works for me and for OfBorg as well now.
Ready for merging I think. Thanks!

**Review by figsoda — APPROVED** [review]

**Pull request merged**